File: fintechplatform-ui/src/main/java/com/fintechplatform/ui/profile/address/ui/AddressUI.java
package com.fintechplatform.ui.profile.address.ui;

import android.content.Context;
import android.content.Intent;

import com.android.volley.toolbox.Volley;
import com.fintechplatform.api.net.NetModule;
import com.fintechplatform.api.profile.api.ProfileAPIModule;
import com.fintechplatform.ui.models.DataAccount;

public class AddressUI {

    private String hostName;
    private DataAccount configuration;

    protected static AddressUI instance;

    public AddressUI(String hostName, DataAccount configuration) {
        this.hostName = hostName;
        this.configuration = configuration;
    }

    protected AddressViewComponent buildAddressComponent(Context context, AddressContract.View view) {
        return DaggerAddressViewComponent.builder()
                .netModule(new NetModule(Volley.newRequestQueue(context),instance.hostName))
                .profileAPIModule(new ProfileAPIModule(instance.hostName))
                .addressPresenterModule(new AddressPresenterModule(view, instance.configuration))
                .build();

    }

    public AddressViewComponent createAddressComponent(Context context, AddressContract.View view){
        return instance.buildAddressComponent(context, view);
    }

    public void start(Context context) {
        instance = this;
        Intent intent = new Intent( context, AddressActivity.class);
        context.startActivity(intent);
    }
}

File: fintechplatform-ui/src/main/java/com/fintechplatform/ui/profile/address/AddressUI.kt
package com.fintechplatform.ui.profile.address

import android.content.Context
import android.content.Intent
import android.os.Bundle
import com.android.volley.toolbox.Volley
import com.fintechplatform.api.net.NetModule
import com.fintechplatform.api.profile.api.IdsDocumentsAPIModule
import com.fintechplatform.api.profile.api.ProfileAPIModule
import com.fintechplatform.ui.models.DataAccount
import com.fintechplatform.ui.profile.address.di.AddressPresenterModule
import com.fintechplatform.ui.profile.address.di.AddressViewComponent
import com.fintechplatform.ui.profile.address.di.DaggerAddressViewComponent


class AddressUI(private val hostName: String, private val dataAccount: DataAccount) {

//
//    fun createAddressComponent(context: Context, view: AddressContract.View): AddressViewComponent {
//        return buildAddressComponent(context, view)
//    }

    fun startActivity(context: Context) {
        val intent = Intent(context, AddressActivity::class.java)
        val args = Bundle()
        args.putString("hostname", hostName)
        args.putParcelable("dataAccount", dataAccount)
        intent.putExtras(args)
        context.startActivity(intent)
    }

    fun createFragment(hostName: String, dataAccount: DataAccount): AddressFragment {
        return AddressFragment.newInstance(hostName, dataAccount)
    }
    object Builder {
        fun buildAddressComponent(context: Context, view: AddressContract.View, hostName: String, dataAccount: DataAccount): AddressViewComponent {
            return DaggerAddressViewComponent.builder()
                    .netModule(NetModule(Volley.newRequestQueue(context), hostName))
                    .profileAPIModule(ProfileAPIModule(hostName))
                    .idsDocumentsAPIModule(IdsDocumentsAPIModule(hostName))
                    .addressPresenterModule(AddressPresenterModule(view, dataAccount))
                    .build()

        }
    }
}
